libraries/ruby_install.rb

```diff
@@ -0,0 +1,247 @@
+#
+# Cookbook Name:: omnibus
+# HWRP:: ruby_install
+#
+# Copyright 2014, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require_relative '_helper'
+
+class Chef
+  class Resource::RubyInstall < Resource::LWRPBase
+    self.resource_name = :ruby_install
+
+    actions :install
+    default_action :install
+
+    attribute :version,     kind_of: String, name_attribute: true
+    attribute :environment, kind_of: Hash, default: {}
+    attribute :patches,     kind_of: Array, default: []
+
```

Comment:
We have separate providers for windows an unix in this file, so that isn't quite right. We were focusing on the linux one for now.